tests/foreman/api/test_host.py

```diff
@@ -1276,6 +1277,68 @@ def test_negative_update_os(self):
             host.read().operatingsystem.read().name, new_os.name)
 
     @run_only_on('sat')
+    @skip_if_bug_open('bugzilla', 1488130)
+    @tier3
+    def test_positive_read_content_source_id(self):
+        """Read the host content_source_id attribute from the read request
+        response
+
+        :id: 0a7fd8d4-1ea8-4b21-8c46-10579644fd11
+
+        :expectedresults: content_source_id is present in GET host request
+            response
+
+        :BZ: 1339613, 1488130
+
+        :CaseLevel: System
+        """
+        proxy = entities.SmartProxy().search(query={
+            'url': 'https://{0}:9090'.format(settings.server.hostname)})[0]
+        host = entities.Host().create()
+        # Update the host with content_source_id
+        response = client.put(
```

Comment:
if you test read functionality in that test case, why don't you use usual `host.upate()` method?

Reply:
nailgun has no content_source field for the moment, and this is the direct call

Reply:
but it can be passed as a parameter for PUT. Does `update` method fail on read operation?

Reply:
@oshtaier  it can be passed as parameter in HTTP POST/PUT but that parameter is never returned back to be read
if you update the host entity with content_source field
```python
class Host(...):
    def __init__(self, server_config=None, **kwargs):
        self._fields = {
            ...
            'content_source': entity_fields.OneToOneField(SmartProxy),
```
and after you want to create a  host 
```python
host = entities.Host().create()
raceback (most recent call last):
  File "/home/dl/.pyenv/versions/sat-6.3/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 2882, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-8-933a92e5f845>", line 1, in <module>
    host = entities.Host(content_source=proxy).create()
  File "/home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/blinker_herald/base.py", line 137, in wrapper
    result = fn(*args, **kwargs)
  File "/home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/nailgun/entities.py", line 3183, in create
    id=self.create_json(create_missing)['id'],
  File "/home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/nailgun/entities.py", line 3272, in read
    result = super(Host, self).read(entity, attrs, ignore, params)
  File "/home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 799, in read
    entity_id = _get_entity_id(field_name, attrs)
  File "/home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/nailgun/entity_mixins.py", line 256, in _get_entity_id
    .format(field_name, (field_name, field_name_id), attrs.keys())
MissingValueError: Cannot find a value for the "content_source" field. Searched for keys named ('content_source', 'content_source_id'), but available keys are [u'comment', u'build_status', u'environment_id', u'managed', u'all_puppetclasses', u'model_id', u'ip', u'provision_method', u'updated_at', u'realm_name', u'compute_resource_id', u'puppet_status', u'sp_mac', u'operatingsystem_name', u'openscap_proxy', u'id', u'openscap_proxy_id', u'disk', u'location_id', u'domain_id', u'build_status_label', u'organization_name', 'host_parameters_attributes', u'location_name', u'ptable_id', u'subnet_id', u'openscap_proxy_name', u'sp_name', u'environment_name', u'architecture_name', u'domain_name', u'capabilities', u'subnet6_id', u'image_id', u'global_status', u'architecture_id', u'build', u'hostgroup_title', u'subnet_name', u'sp_ip', u'model_name', u'puppet_proxy_id', u'global_status_label', u'owner_id', u'operatingsystem_id', u'sp_subnet_id', u'puppet_ca_proxy_name', u'config_groups', u'name', u'permissions', u'compute_profile_name', u'interfaces', u'certname', u'image_name', u'organization_id', u'image_file', u'mac', u'ip6', u'puppet_ca_proxy_id', u'puppetclasses', u'puppet_ca_proxy', u'last_report', u'puppet_proxy', u'medium_name', u'last_compile', u'uuid', u'pxe_loader', u'ptable_name', u'medium_id', u'compute_resource_name', u'created_at', u'enabled', u'hostgroup_id', u'subnet6_name', u'hostgroup_name', u'use_image', u'compute_profile_id', u'all_parameters', u'owner_type', u'realm_id', u'installed_at', u'puppet_proxy_name'].
```
so in nailgun the post passed but reading the entity failed and create function failed with in the read function because that file is (content_source_id is not returned)

we have only on way to update that field is to use direct HTTP PUT
to verify read use the nailgun read_json to see if the key is there

Reply:
@oshtaier responding to your question , any operation that use read will fail on read operation

Reply:
> nailgun has no content_source field for the moment, and this is the direct call, we cannot add this field in nailgun as not returned at all for the moment, an will break host entity

That's basically a reason why i postponed API part of the issue whilst covered other interfaces. And i'm still not sure whether it's a good idea to introduce such test with direct api calls which demands refactor, maybe it's better to cover it when the issue is on_qa?

Reply:
@abalakh , this test should never be refactored even when that field appear , as it has to verify directly the return of PUT

Reply:
@ldjebran and what's your concern? `update()` will verify the field is present in output

Reply:
@abalakh update will verify all the fields and fail if any does not exist even before it reach the verification of the current field, **we want to verify only this Field** if the test fail it should fail only for this field.